src/main/java/gov/cabinetoffice/gapfindapiadmin/security/JwtAuthorisationFilter.java

```diff
@@ -0,0 +1,62 @@
+package gov.cabinetoffice.gapfindapiadmin.security;
+
+import com.auth0.jwt.interfaces.DecodedJWT;
+import gov.cabinetoffice.gapfindapiadmin.config.OneLoginConfig;
+import gov.cabinetoffice.gapfindapiadmin.exceptions.UnauthorizedException;
+import gov.cabinetoffice.gapfindapiadmin.models.JwtPayload;
+import gov.cabinetoffice.gapfindapiadmin.services.JwtService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.springframework.util.ObjectUtils.isEmpty;
+
+@RequiredArgsConstructor
+public class JwtAuthorisationFilter extends OncePerRequestFilter {
+
+    private final JwtService jwtService;
+
+    private final OneLoginConfig oneLoginConfig;
+
+
+    @Override
+    public void doFilterInternal(HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull FilterChain filterChain) throws ServletException, IOException {
+        final String header = request.getHeader("Authorization");
+        if (isEmpty(header) || !header.startsWith("Bearer ")) {
+            throw new UnauthorizedException("Expected Authorization header not provided");
+        }
+
+        final String jwtBase64 = header.replace("Bearer ", "");
+
+        final DecodedJWT decodedJWT = this.jwtService.verifyToken(jwtBase64);
+
+        JwtPayload JWTPayload;
+        if (oneLoginConfig.isOneLoginEnabled()) {
+            JWTPayload = this.jwtService.getPayloadFromJwtV2(decodedJWT);
+        }
+        else {
```

Comment:
I actually think for our purpose we maybe won't need the COLA stuff here as all users will eventually need to migrate to OneLogin and if they don't then they can't use the system so we could potentially remove that part of the code and simplify/clean things up a bit. 

Keen for your thoughts @iaincooper-tco

Reply:
OneLogin is going live at the end of this month, so we are never going to have a live version of the API with COLA. We don't need to include COLA compatibility with our work.